ManagementPacks/Community.DataOnDemand/Scripts/Watch-TcpConnections.ps1

```diff
@@ -0,0 +1,539 @@
+
+                    // Ignore listening state
+                    TcpState state = (TcpState)ipv4.state;
+                    if (state == TcpState.Listen) { continue; }
+
+                    // Ignore loopback connections
+                    IPEndPoint remote = new IPEndPoint(new IPAddress(ipv4.remoteAddr), ipv4.remotePort[0] << 8 | ipv4.remotePort[1]);
+                    if (IPAddress.IsLoopback(remote.Address)) { continue; }
+                    IPEndPoint local = new IPEndPoint(new IPAddress(ipv4.localAddr), ipv4.localPort[0] << 8 | ipv4.localPort[1]);
+
+                    // ...so we are interested in this one
+                    TcpConnectionV01 conn = new TcpConnectionV01(
+                        computername,
+                        local,
+                        remote,
+                        owningPid,
+                        state,
+                        first);
+
+                    TcpConnectionV01 existingConn;
+                    if (connsByKey.TryGetValue(conn.Key, out existingConn))
+                    {
+                        // We have seen this one before, update the state in the existing entry
+                        if (existingConn.CurrentState == conn.CurrentState)
+                        {
+                            existingConn.StateSamples[existingConn.StateSamples.Count - 1].Count++;
+                        }
+                        else
+                        {
+                            existingConn.CurrentState = conn.CurrentState;
+                            existingConn.StateSamples.Add(new TcpStateSampleV01(conn.CurrentState));
+                        }
+
+                        // Note that we have seen this one again
+                        connsNotSeen.Remove(conn.Key);
+                    }
+                    else
+                    {
+                        // Create a new entry
+                        conn.StateSamples.Add(new TcpStateSampleV01(conn.CurrentState));
+                        conn.TimeFirstSeen = conn.StateSamples[0].TimeSampled;
+                        conn.ProcessName = "Unknown";
+                        conn.ProcessDescription = "Unknown";
+                        connsByKey[conn.Key] = conn;
+                    }
+                }
+
+                // Now go through all the ones we didn't see this time around and mark as gone
+                foreach (string keyGone in connsNotSeen)
+                {
+                    TcpConnectionV01 conn = connsByKey[keyGone];
+                    if (!conn.IsGone)
+                    {
+                        conn.IsGone = true;
+                        conn.StateSamples.Add(new TcpStateSampleV01("|GONE|"));
+                        conn.TimeFirstGone = conn.StateSamples[conn.StateSamples.Count - 1].TimeSampled;
+                    }
+                }
+
+                // Wait one second
+                Thread.Sleep(TimeSpan.FromSeconds(1));
```

Comment:
Why? (Is this something like always wanting to end >= `nSeconds+1` for some reason?)